core/integration/services_test.go

```diff
@@ -571,15 +612,135 @@ func TestContainerExecServicesChained(t *testing.T) {
+		}, dagger.ContainerWithExecOpts{
+			ExperimentalPrivilegedNesting: true,
+		}).
+		Stdout(ctx)
+	require.NoError(t, err)
+	require.Equal(t, content, fileContent)
+}
+
+func TestContainerExecServicesNestedHTTP(t *testing.T) {
+	t.Parallel()
+
+	c, ctx := connect(t)
+	defer c.Close()
+
+	thisRepoPath, err := filepath.Abs("../..")
+	require.NoError(t, err)
+
+	code := c.Host().Directory(thisRepoPath, dagger.HostDirectoryOpts{
+		Include: []string{"core/integration/testdata/nested/", "sdk/go/", "go.mod", "go.sum"},
+	})
+
+	content := identity.NewID()
+	srv, svcURL := httpService(ctx, t, c, content)
+
+	// NB: currently maxes out around 8-9 due to 256 character limit on DNS
+	// search domain config
+	depth := 5
+
+	fileContent, err := c.Container().
+		From("golang:1.20.6-alpine").
+		WithServiceBinding("www", srv).
+		WithMountedDirectory("/src", code).
+		WithWorkdir("/src").
+		WithMountedCache("/go/pkg/mod", c.CacheVolume("go-mod")).
+		WithEnvVariable("GOMODCACHE", "/go/pkg/mod").
+		WithMountedCache("/go/build-cache", c.CacheVolume("go-build")).
+		WithEnvVariable("GOCACHE", "/go/build-cache").
+		WithExec([]string{
+			"go", "run", "./core/integration/testdata/nested/",
+			"http", strconv.Itoa(depth), svcURL,
+		}, dagger.ContainerWithExecOpts{
+			ExperimentalPrivilegedNesting: true,
+		}).
+		Stdout(ctx)
+	require.NoError(t, err)
+	require.Equal(t, content, fileContent)
+}
+
+func TestContainerExecServicesNestedGit(t *testing.T) {
 	t.Parallel()
 
-	checkNotDisabled(t, engine.ServicesDNSEnvName)
+	c, ctx := connect(t)
+	defer c.Close()
+
+	thisRepoPath, err := filepath.Abs("../..")
+	require.NoError(t, err)
+
+	code := c.Host().Directory(thisRepoPath, dagger.HostDirectoryOpts{
+		Include: []string{"core/integration/testdata/nested/", "sdk/go/", "go.mod", "go.sum"},
+	})
+
+	content := identity.NewID()
+	srv, svcURL := gitService(ctx, t, c, c.Directory().WithNewFile("/index.html", content))
+
+	// NB: currently maxes out around 8-9 due to 256 character limit on DNS
+	// search domain config
+	depth := 5
+
+	fileContent, err := c.Container().
+		From("golang:1.20.6-alpine").
+		WithServiceBinding("www", srv).
+		WithMountedDirectory("/src", code).
+		WithWorkdir("/src").
+		WithMountedCache("/go/pkg/mod", c.CacheVolume("go-mod")).
+		WithEnvVariable("GOMODCACHE", "/go/pkg/mod").
+		WithMountedCache("/go/build-cache", c.CacheVolume("go-build")).
+		WithEnvVariable("GOCACHE", "/go/build-cache").
+		WithExec([]string{
+			"go", "run", "./core/integration/testdata/nested/",
+			"git", strconv.Itoa(depth), svcURL,
+		}, dagger.ContainerWithExecOpts{
+			ExperimentalPrivilegedNesting: true,
+		}).
+		Stdout(ctx)
+	require.NoError(t, err)
+	require.Equal(t, content, fileContent)
+}
+
+func TestContainerBuildService(t *testing.T) {
+	t.Parallel()
 
 	c, ctx := connect(t)
 	defer c.Close()
 
 	t.Run("building with service dependency", func(t *testing.T) {
+		t.Skip("this no longer works, and it's kind of weird that it ever did")
```

Comment:
@vito I agree it's weird this ever worked 😄 You good with just removing these skipped ones now?

Reply:
Yeah sounds good - if you or I don't remove them on this branch I'll just remove them in whatever becomes the services v2 PR. :+1:

Reply:
removed in https://github.com/dagger/dagger/pull/5415/commits/c07b171a7da7efcdb478c92343e8011017bede94